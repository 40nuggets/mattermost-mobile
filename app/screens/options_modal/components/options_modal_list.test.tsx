// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithIntl} from '@test/intl-test-helper';

import OptionModalListIOS from './options_modal_list';

describe('OptionModalList', () => {
    const baseProps = {
        items: [{
            action: jest.fn(),
            text: {
                id: 'mobile.file_upload.camera',
                defaultMessage: 'Take Photo or Video',
            },
            icon: 'camera',
        }, {
            action: jest.fn(),
            text: {
                id: 'mobile.file_upload.library',
                defaultMessage: 'Photo Library',
            },
            icon: 'photo',
        }],
        onCancelPress: jest.fn(),
        title: 'test',
    };

    test('should match snapshot ', async () => {
        const wrapper = renderWithIntl(
            <OptionModalListIOS {...baseProps}/>,
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
