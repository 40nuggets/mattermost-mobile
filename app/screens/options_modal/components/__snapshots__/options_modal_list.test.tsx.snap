// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionModalList should match snapshot  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "maxWidth": 450,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "white",
            "borderRadius": 12,
            "marginBottom": 20,
            "marginHorizontal": 20,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 15,
            },
            Object {
              "borderBottomColor": "rgba(0, 0, 0, 0.1)",
              "borderBottomWidth": 1,
            },
          ]
        }
      >
        <Text>
          test
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "rgba(0, 0, 0, 0.1)",
              "borderBottomWidth": 1,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 16,
                },
                undefined,
                false,
              ]
            }
          >
            Take Photo or Video
          </Text>
          <Icon
            name="camera"
            size={18}
            style={
              Object {
                "color": "#7f8180",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            false,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 16,
                },
                undefined,
                false,
              ]
            }
          >
            Photo Library
          </Text>
          <Icon
            name="photo"
            size={18}
            style={
              Object {
                "color": "#7f8180",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "borderRadius": 12,
          "marginBottom": 20,
          "marginHorizontal": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#CC3239",
              "flex": 1,
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
